<!--
  ~ Copyright (c) 2018-2019 VMware, Inc. All Rights Reserved.
  ~ This software is released under MIT license.
  ~ The full license information can be found in LICENSE in the root directory of this project.
  -->

<p>The <code>Collator</code> class should not be used without an explicit, user-preferred locale argument.</p>

<h2>Noncompliant Code Example</h2>
<pre>
  Collator.getInstance()
</pre>
<h2>Compliant Solution</h2>
<pre>
  Locale userPreferredLocale = Locale.forLanguageTag("zh-Hans");
  ...
  Collator.getInstance(userPreferredLocale)
</pre>
